command/agent/command.go

```diff
@@ -427,10 +432,130 @@ func (c *Command) startAgent(config *Config, agent *Agent,
 	if config.Discover != "" {
 		c.Ui.Info(fmt.Sprintf("  mDNS cluster: %s", config.Discover))
 	}
+
 	return ipc
 }
 
 // startupJoin is invoked to handle any joins specified to take place at start time
+func (c *Command) startupJoinSRV(config *Config, agent *Agent) error {
+	if len(config.JoinSRV) == 0 {
+		return nil
+	}
+
+	c.Ui.Output(fmt.Sprintf("Joining cluster via SRV...(replay: %v)", config.ReplayOnJoin))
+	n, err := c.joinSRV(agent, config.ReplayOnJoin, config.JoinSRV)
+	if err != nil {
+		return err
+	}
+
+	if n == 0 {
+		return errors.New("Failed to join any hosts via SRV")
+	}
+
+	c.Ui.Info(fmt.Sprintf("Join completed. Synced with %d initial agents", n))
+	return nil
+
+}
+
+// retryJoinSRV is invoked to handle joins with retries. This runs until at least a
+// single successful join or RetryMaxAttempts is reached
+func (c *Command) retryJoinSRV(config *Config, agent *Agent, errCh chan struct{}) {
+	// Quit fast if there is no nodes to join
+	if len(config.RetryJoinSRV) == 0 {
+		return
+	}
+
+	// Track the number of join attempts
+	attempt := 0
+	for {
+		// Try to perform the join
+		n, err := c.joinSRV(agent, config.ReplayOnJoin, config.RetryJoinSRV)
+
+		if err != nil {
+			c.logger.Printf("[ERR] agent: Failed to join via SRV: %v", err)
+		}
+
+		if err == nil && n > 0 {
+			c.logger.Printf("[INFO] agent: Join completed. Synced with %d initial agents", n)
```

Comment:
What's the use case for continuous polling? Once you join, the cluster takes care of membership and a new guy can bootstrap with one of the agents in the srv record. 

> On Sep 14, 2015, at 9:11 PM, Dale Hamel <notifications@github.com> wrote:
> 
> In command/agent/command.go:
> 
> > +	if len(config.RetryJoinSRV) == 0 {
> > +		return
> > +	}
> > +
> > +	// Track the number of join attempts
> > +	attempt := 0
> > +	for {
> > +		// Try to perform the join
> > +		n, err := c.joinSRV(agent, config.ReplayOnJoin, config.RetryJoinSRV)
> > +
> > +		if err != nil {
> > +			c.logger.Printf("[ERR] agent: Failed to join via SRV: %v", err)
> > +		}
> > +
> > +		if err == nil && n > 0 {
> > +			c.logger.Printf("[INFO] agent: Join completed. Synced with %d initial agents", n)
> I'm wondering if it would make sense to make the return conditional, to provide something like a 'max trys = -1' to just keep polling, even after the cluster has been joined.
> 
> We have a use case for continuous polling.
> 
> —
> Reply to this email directly or view it on GitHub.
>